src/app/(layout-with-navigation)/personas/_helpers/getPersonaFoldedSteps.ts

```diff
@@ -0,0 +1,62 @@
+import getQuestionsOfMosaic from '@/publicodes-state/helpers/getQuestionsOfMosaic'
+import { safeGetSituation } from '@/publicodes-state/helpers/safeGetSituation'
+import {
+  DottedName,
+  Engine,
+  NGCEvaluatedNode,
+  NGCRuleNode,
+} from '@/publicodes-state/types'
+import { Situation } from '@/types/simulation'
+import { fixSituationWithPartialMosaic } from './fixSituationWithPartialMosaic'
+
+export const getPersonaFoldedSteps = (
+  situation: Situation,
+  everyMosaic: DottedName[],
+  everyMosaicChildren: DottedName[],
+  everyQuestions: DottedName[],
+  everyRules: DottedName[],
+  pristineEngine: Engine,
+  safeGetRule: (rule: DottedName) => NGCRuleNode | null,
+  safeEvaluate: (rule: DottedName) => NGCEvaluatedNode | null
+) => {
+  const personaSituation = fixSituationWithPartialMosaic(
+    situation,
+    everyMosaic,
+    everyMosaicChildren,
+    safeGetRule,
+    safeEvaluate
+  )
+
+  const safeSituation = safeGetSituation({
+    situation: personaSituation,
+    everyRules,
+  })
+
+  pristineEngine.setSituation(safeSituation)
+
+  const missingVariables = Object.keys(
```

Comment:
```suggestion
  // The persona folded steps are obtained by getting the missing variables
  const personaFoldedSteps = Object.keys(
```
et supprimer la ligne du dessous (commentaire à amender en fonction)